lib/Wikia/src/Persistence/User/Preferences/PreferencePersistenceMySQL.php

```diff
@@ -109,9 +109,18 @@ public static function createTuplesFromPreferences( $userId, array $preferences
 	public function get( $userId ) {
 		$userId = intval( $userId );
 		$cond = [ self::UP_USER => $userId ];
-		if ( !empty( $this->whiteList ) ) {
-			$cond[] = '`' . self::UP_PROPERTY . '` IN (' . $this->slave->makeList($this->whiteList['literals']) . ') ' .
-				'OR `' . self::UP_PROPERTY . "` REGEXP " . $this->slave->addQuotes($this->whiteList['regex']);
+
+		if ( is_array( $this->whiteList ) ) {
+			$orConds = [ ];
+			if ( is_array( $this->whiteList[ 'literals' ] ) ) {
+				$orConds[ ] = '`' . self::UP_PROPERTY . '` IN (' . $this->slave->makeList( $this->whiteList[ 'literals' ] ) . ')';
+			}
+			//white list regex is combined regexes into one string
+			if ( !empty( $this->whiteList[ 'regex' ] ) ) {
+				$orConds[ ] = '`' . self::UP_PROPERTY . "` REGEXP " . $this->slave->addQuotes( $this->whiteList[ 'regex' ] );
```

Comment:
Confusion is now gone after offline chat with @sqreek - thanks for giving a bit of context here :)